iri-commons/src/main/java/no/sikt/nva/iri/PublicationChannelId.java

```diff
@@ -0,0 +1,136 @@
+package no.sikt.nva.iri;
+
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.net.URI;
+import java.time.DateTimeException;
+import java.time.Year;
+import java.util.UUID;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+public interface PublicationChannelId {
+
+    String INVALID_HOST = "Provided host is invalid";
+    String INVALID_PATH_LENGTH =
+            "%s must have a path matching /{CHANNELS-PATH}/{TYPE}/{UUID}/{YYYY}";
+    String INVALID_PATH_START = "%s must have a path starting /{CHANNELS-PATH}";
+    String INVALID_TYPE = "%s must contain a valid type";
+    String INVALID_URI = "Channel URI should have https scheme";
+    String INVALID_UUID = "%s must contain a valid UUID";
+    String INVALID_YEAR = "%s must contain a valid year";
+    String PATH_ELEMENT_ONE = "publication-channels-v2";
+    String PATH_TEMPLATE = "/%s/%s/%s/%s";
+    String SLASH = "/";
+    String TOO_EARLY_YEAR = "%s must have a year after 2004";
+    Year CHANNEL_REGISTRY_FOUNDATION_YEAR = Year.of(2004);
+    int EXPECTED_PATH_LENGTH = 5;
+    String HTTPS_SCHEME = "https";
+    String SCHEME_SEPARATOR = "://";
+    String PLACEHOLDER_QUALIFIED_HOST = "https://example.org";
+    String QUOTED_STRING = "\"";
+    String EMPTY_STRING = "";
+
+    @JsonCreator
+    static PublicationChannelId create(String json) {
+        var uri = URI.create(PLACEHOLDER_QUALIFIED_HOST + json.replace(QUOTED_STRING, EMPTY_STRING));
+        return from(uri);
+    }
+
+    static PublicationChannelId from(URI uri) {
+        if (nonNull(uri) && uri.toString().contains(ChannelType.PUBLISHER.getType())) {
+            return PublisherId.from(uri);
+        } else if (nonNull(uri) && uri.toString().contains(ChannelType.SERIAL_PUBLICATION.getType())) {
+            return SerialPublicationId.from(uri);
+        } else {
+            throw new IllegalArgumentException("Encountered URI that is not a valid publication channel id");
+        }
+    }
```

Comment:
If we want to be very strict, should we validate that ChannelType.PUBLISHER.getType() and ChannelType.SERIAL_PUBLICATION.getType() are in right places? It seems that we do it downstream :)